WordPress/WordPressTest/ContextManagerMock+AutomaticTeardown.swift

```diff
@@ -0,0 +1,72 @@
+import XCTest
+
+@testable import WordPress
+
+extension ContextManagerMock {
+    /// Override `ContextManager.shared` with the mock instance, and un-override it when
+    /// given `testCase` finishes.
+    ///
+    /// This method should be called during test set up, typically from `XCTestCase` subclass's
+    /// `setUp` method.
+    ///
+    /// - Parameter testCase: The test case to wait for.
+    @objc func useAsSharedInstance(untilTestFinished testCase: XCTestCase) {
+        // Create the test observer singleton to add it to `XCTestObservationCenter`.
+        _ = AutomaticTeardownTestObserver.instance
+
+        let original = ContextManager.overrideInstance
+        ContextManager.internalSharedInstance()
+        ContextManager.overrideSharedInstance(self)
+
+        // This closure is going to be called by the test observer below when
+        // the test case finishes. The reason a combination of storing a closure in
+        // `XCTestCase` instance and a dedicated `XCTestObservation` implementation
+        // is used here is, we need to make sure tearing down the `ContextManager`
+        // mock instance happens _after_ test finishes execution.
+        //
+        // XCTest does have an API allowing us to add our own teardown block:
+        // `XCTestCase.addTeardownBlock`, but the added block is called _before_
+        // the test case's `tearDown` override method. Which means if this official
+        // API is used here instead, `ContextManager.shared` references two different
+        // objects during test execution: the mock instance before `tearDown`, or the
+        // real singleton during `tearDown`, which isn't ideal.
```

Comment:
👍 

Alternative suggested (but I'm fine with keeping the current wording as well):
```suggestion
        // XCTest does have an API allowing us to add our own teardown block:
        // `XCTestCase.addTeardownBlock`, but the added block is called _before_
        // the test case's `tearDown` override method. Which means if this official
        // API is used here instead, then calling `ContextManager.shared` (explicitly
        // or indirectly) during the test's `tearDown` would call the real singleton
        // instead of the mock, which isn't ideal and could have unintended side effect.
```

Reply:
Love the wording in your suggestion. Thanks!